src/sentry/api/endpoints/monitor_stats.py

```diff
@@ -16,24 +16,40 @@ def get(self, request: Request, project, monitor) -> Response:
         args = self._parse_args(request)
 
         stats = OrderedDict()
+        duration_stats = OrderedDict()
         current = tsdb.normalize_to_epoch(args["start"], args["rollup"])
         end = tsdb.normalize_to_epoch(args["end"], args["rollup"])
         while current <= end:
             stats[current] = {CheckInStatus.OK: 0, CheckInStatus.ERROR: 0}
+            duration_stats[current] = {"sum": 0, "num_checkins": 0}
             current += args["rollup"]
 
         history = MonitorCheckIn.objects.filter(
             monitor=monitor,
             status__in=[CheckInStatus.OK, CheckInStatus.ERROR],
             date_added__gt=args["start"],
             date_added__lte=args["end"],
-        ).values_list("date_added", "status")
-        for datetime, status in history.iterator():
-            stats[tsdb.normalize_to_epoch(datetime, args["rollup"])][status] += 1
-
-        return Response(
-            [
-                {"ts": ts, "ok": data[CheckInStatus.OK], "error": data[CheckInStatus.ERROR]}
-                for ts, data in stats.items()
-            ]
-        )
+        ).values_list("date_added", "status", "duration")
+        for datetime, status, duration in history.iterator():
```

Comment:
This endpoint is getting hard to read. Could you either add comments or break stuff out into helper functions?

Reply:
will do